tests/schema/__app/generated/schema.graphql

```diff
@@ -52,7 +52,7 @@ input DateTimeFilter {
   in: [DateTime!]
   lt: DateTime
   lte: DateTime
-  not: DateTime
+  not: NestedDateTimeFilter
```

Comment:
Talked with @timsuchanek. This change reflects dmmf changes made to support https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/filtering#case-insensitive-filtering.

Larger example of what is going on:

```gql
input IntFilter {
  equals: Int
  in: [Int!]
  notIn: [Int!]
  lt: Int
  lte: Int
  gt: Int
  gte: Int
  not: NestedIntFilter
}
input NestedIntFilter {
  equals: Int
  in: [Int!]
  notIn: [Int!]
  lt: Int
  lte: Int
  gt: Int
  gte: Int
  not: NestedIntFilter
}
```

Our conclusion is that at the gql layer we don't need the indirection. We can just have `IntFilter` recurse into itself.

New issue to support case insensitivity in gql layer here https://github.com/graphql-nexus/nexus-plugin-prisma/issues/806.

Reply:
Few points from Dom

- top-level type in dmmf is where `mode` exists
- `mode` is not supported in the nested types that recurse
- this means the types meaningfully diverge
- we cannot put `mode` on every level and have that be usable at the prisma client level
- so to support `mode` we will have to double up our filter types
- so we might as well start now in prep for that later
- there are no features with regards to non-string-ish filters currently
- but there might be in the future, and by way of consistency of pattern it is arguably simpler to have all filter types have nested types.

conclusion is that we will not collapse anything

Instead we need to document this new pattern in the schema contributions docs.